### ISSUE TITLE
put the monitor metric into default filename for ModelCheckpoint

### ISSUE THREAD

**Issue by VDFaller** [Lightning-AI/pytorch-lightning#20397]
### Description & Motivation

Tiny annoyance, but wouldn't it make sense to put the monitor value in the default value of the filename so it's not just `epoch-X-step-y` by default?
 

### Pitch

Couldn't something like this work [here](https://github.com/Lightning-AI/pytorch-lightning/blob/master/src/lightning/pytorch/callbacks/model_checkpoint.py#L533-L542)?  

```python
    def _format_checkpoint_name(
        self,
        filename: Optional[str],
        metrics: Dict[str, Tensor],
        prefix: str = "",
        auto_insert_metric_name: bool = True,
    ) -> str:
        if not filename:
            if self.monitor is not None and self.monitor in metrics:
                filename = "{epoch}" + self.CHECKPOINT_JOIN_CHAR + "{step}" + self.CHECKPOINT_JOIN_CHAR + f"{{{self.monitor}}}"
            # filename is not set, use default name
            else:
                filename = "{epoch}" + self.CHECKPOINT_JOIN_CHAR + "{step}"

```

### Additional context

Happy to put it in MR

**Comment by lantiga** [comment]
Thanks @VDFaller, can you briefly elaborate on what's the itch you'd like to scratch in your workflow?

**Comment by VDFaller** [comment]
@lantiga no problem. 
I have developers that give me checkpoints in a folder with no context sometimes. And they forget to put the val loss (or whatever) in their filename. So I have to rerun their validation to see which is which. 

Maybe it's stored elsewhere than the filename that I could extract? Am I making it too difficult? 

Like I said, tiny annoyance.
